docs/advanced-features/react-18/server-components.md

```diff
@@ -0,0 +1,124 @@
+# React Server Components (Alpha)
+
+Server Components allow us to render React components on the server. This is fundamentally different from server-side rendering where you're pre-generating HTML on the server. With Server Components, there's **zero client-side JavaScript needed,** making page rendering faster. This improves the user experience of your application, pairing the best parts of server-rendering with client-side interactivity.
+
+### Enable React Server Components
+
+To use React Server Components, ensure you have React 18 installed. Then, turn on the `concurrentFeatures` and `serverComponents` options in `next.config.js`:
+
+```jsx
+// next.config.js
+module.exports = {
+  experimental: {
+    concurrentFeatures: true,
+    serverComponents: true,
+  },
+}
+```
+
+Next, if you already have customized `pages/_document` component, you need to remove the `getInitialProps` static method and the `getServerSideProps` export if there’s any, otherwise it won't work with server components. If no custom Document component is provided, Next.js will fallback to a basic functional document component. Check [next/document support](#next/document) for more details.
+
+Then, you can start using React Server Components. [See our example](https://github.com/vercel/next-rsc-demo) for more information.
+
+### Server Components Convention
+
+To run a component on the server, append `.server.js` to the end of the filename. For example `./pages/home.server.js` is a Server Component.
+
+For client components, add `.client.js`. For example, `./components/avatar.client.js`.
+
+You can then import other server or client components from any server component. Note: a server component **can not** be imported by a client component. Components without "server/client" extensions will be treated as "universal component" and can be used and rendered by both sides, depending on where it is imported. For example:
+
+```jsx
+// pages/home.server.js
+
+import { Suspense } from 'react'
+
+import Profile from '../components/profile.server'
+import Content from '../components/content.client'
+
+export default function Home() {
+  return (
+    <div>
+      <h1>Welcome to React Server Components</h1>
+      <Suspense fallback={'Loading...'}>
+        <Profile />
+      </Suspense>
+      <Content />
+    </div>
+  )
+}
+```
+
+The `<Home>` and `<Profile>` components will always be server-side rendered and streamed to the client, and will not be included by the client runtime. However `<Content>` will still be hydrated on the client-side, like normal React components.
+
+To see a full example, check out [link to the demo and repository](https://github.com/vercel/next-rsc-demo).
+
```

Comment:
Out of curiosity, what are the implications of using non-default exports? Will this work the same way?

```ts
import { Profile } from '../components/profile.server'
import { Content } from '../components/content.client'
```

Asking because I know repos using [import/no-default-export](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-default-export.md) rule everywhere except `pages` directory.

Reply:
This is not supported so far due to the early implementation but would be great to have

Reply:
I see. What do you think of mentioning this limitation in a quick sentence? Might be a common expectation to have.